test/new_tests/test_query_apply.py

```diff
@@ -508,3 +509,57 @@ def _no_items_have_been_applied(self):
             assert bins["name"] != "aerospike"
         else:
             assert bins["name"] == "no_set_name"
+
+    def test_fail_if_operations_exist_before_apply(self):
+        """
+        Invoke query.apply() with a stream udf.
+        arguments contain a serialized set.
+        """
+        ops = [
+            operations.increment("testBinName", 1)
+        ]
+
+        try:
+            query_results = (
+                self.as_connection.query(
+                    "test",
+                    "demo",
+                )
+                .add_ops(ops)
+                .apply(
+                    "query_apply_parameters",
+                    "query_params",
+                    [["age", 5], pickle.dumps({"lary", "quinton", "julie", "mark"})],
+                )
+                .results()
+            )
+            assert "Param error should be thrown" == "This code should not be executed"
+        except e.ParamError:
+            pass
+
+    def test_fail_if_UDF_exists_before_add_ops(self):
+        """
+        Invoke query.apply() with a stream udf.
+        arguments contain a serialized set.
+        """
+        ops = [
+            operations.increment("testBinName", 1)
+        ]
+
+        try:
+            query_results = (
+                self.as_connection.query(
+                    "test",
+                    "demo",
+                )
+                .apply(
+                    "query_apply_parameters",
+                    "query_params",
+                    [["age", 5], pickle.dumps({"lary", "quinton", "julie", "mark"})],
+                )
+                .add_ops(ops)
+                .results()
+            )
+            assert "Param error should be thrown" == "This code should not be executed"
+        except e.ParamError:
+            pass
```

Comment:
I don't think we need a second test case since the first one already tests query.results()
```suggestion
    def test_fail_if_UDF_exists_before_add_ops(self):
        """
        Invoke query.apply() with a stream udf.
        arguments contain a serialized set.
        """
        ops = [
            operations.increment("testBinName", 1)
        ]

        try:
            query_results = (
                self.as_connection.query(
                    "test",
                    "demo",
                )
                .apply(
                    "query_apply_parameters",
                    "query_params",
                    [["age", 5], pickle.dumps({"lary", "quinton", "julie", "mark"})],
                )
                .add_ops(ops)
                .results()
            )
            assert "Param error should be thrown" == "This code should not be executed"
        except e.ParamError:
            pass
```

Reply:
I included the wrong test cases in this branch, I removed them and added the correct ones, so these comments should not be relevant anymore.